polus-image-assembler-plugin/src/main.py

```diff
@@ -256,6 +261,9 @@ def main(imgPath: pathlib.Path,
         logger.info(f'Unable to infer pattern, defaulting to: .*')
         fp = filepattern.FilePattern(imgPath,'.*')
 
+    # get z depth
+    depth = BioReader(next(imgPath.iterdir())).z
```

Comment:
You should do this inside of a `with` block. What you are functionally doing here is opening a file and leaving it open. Good coding practice would be to create a BioReader object, read the depth, and then close it. We have run into problems in the past with too many files opens on an OS, and it could be related to this. Using the `with` block opens the image, allows to perform your operations, and then closes the file for you once the code exits the `with` block.